src/jit/morph.cpp

```diff
@@ -7038,20 +7113,75 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
         return false;
     }
 
+    const unsigned maxRegArgs = MAX_REG_ARG;
+
     // If we reached here means that callee has only those argument types which can be passed in
     // a register and if passed on stack will occupy exactly one stack slot in out-going arg area.
-    // If we are passing args on stack for callee and it has more args passed on stack than
-    // caller, then fast tail call cannot be performed.
+    // If we are passing args on stack for the callee and it has more args passed on stack than
+    // the caller, then fast tail call cannot be performed.
     //
     // Note that the GC'ness of on stack args need not match since the arg setup area is marked
     // as non-interruptible for fast tail calls.
-    if ((nCalleeArgs > MAX_REG_ARG) && (nCallerArgs < nCalleeArgs))
+
+#ifdef WINDOWS_AMD64_ABI
+    // x64 Windows: If we have more callee registers used than MAX_REG_ARG, then 
+    // make sure the callee's incoming arguments is less than the caller's
+    if ((((calleeArgRegCount + calleeFloatArgRegCount) > maxRegArgs) && 
+        ((callerArgRegCount + callerFloatArgRegCount) > (callerArgRegCount + callerFloatArgRegCount)))
     {
+        JITDUMP("Will not fastTailCall ((calleeArgRegCount + calleeFloatArgRegCount) > maxRegArgs) && (callerArgRegCount > calleeArgRegCount)");
+        return false;
+    }
+
+#elif (_TARGET_AMD64_ && UNIX_AMD64_ABI) || _TARGET_ARM64_
+
+    // For *nix Amd64 and Arm64 check to see if all arguments for the callee
+    // and caller are passing in registers. If not make sure that the stack
+    // size for the callee is at less than or equal to the caller's.
```

Comment:
>is at less [](start = 27, length = 10)

> "... is at less ..."

Should the "at" be removed? I might rephrase to say:

```
If not, ensure that the outgoing argument stack size requirement for the callee is less than or equal to the caller's entire stack frame usage.
```

(I assume we care about the caller's entire stack frame usage, not the caller's argument stack space usage, which would be less.)